packages/x-data-grid/src/components/columnSelection/GridCellCheckboxRenderer.tsx

```diff
@@ -56,7 +57,16 @@ const GridCellCheckboxForwardRef = forwardRef<HTMLInputElement, GridRenderCellPa
       },
     );
 
-    const disabled = !isSelectable;
+    const [disabled, setDisabled] = React.useState(
+      isSelectable === undefined ? undefined : !isSelectable,
+    );
+
+    useEnhancedEffect(() => {
```

Comment:
You should be able to do the `setDisabled` in the render to avoid the double render